video/out/wayland_common.c

```diff
@@ -2648,6 +2729,22 @@ static void set_color_management(struct vo_wayland_state *wl)
 #endif
 }
 
+static void set_color_representation(struct vo_wayland_state *wl)
+{
+    struct mp_image_params target_params = vo_get_target_params(wl->vo);
+    if (!wl->color_representation_surface || pl_color_repr_equal(&target_params.repr, &wl->target_params.repr))
+        return;
+
+    struct pl_color_repr *repr = &target_params.repr;
+    int alpha = wl->alpha_map[repr->alpha];
+    int coefficients = wl->coefficients_map[repr->sys];
+    int range = repr->levels == PL_COLOR_LEVELS_FULL ? wl->range_map[repr->sys] :
+                                wl->range_map[repr->sys + PL_COLOR_SYSTEM_COUNT];
+    if (coefficients && range)
+        wp_color_representation_surface_v1_set_coefficients_and_range(wl->color_representation_surface, coefficients, range);
+    wp_color_representation_surface_v1_set_alpha_mode(wl->color_representation_surface, alpha);
```

Comment:
The compositor might not support the required alpha mode (not sure if you want or need a fallback for this)
```suggestion
    if (alpha)
        wp_color_representation_surface_v1_set_alpha_mode(wl->color_representation_surface, alpha);
```